demos/text-field.html

```diff
@@ -533,6 +537,18 @@ <h2>Full-Width Text Field and Textarea</h2>
           });
         });
 
+        document.getElementById('required-leading-trailing').addEventListener('change', function(evt) {
+          var target = evt.target;
+          tfBoxLeadingEl.querySelector('.mdc-text-field__input').required = target.checked;
+          tfBoxTrailingEl.querySelector('.mdc-text-field__input').required = target.checked;
+          tfOutlinedLeadingEl.querySelector('.mdc-text-field__input').required = target.checked;
+          tfOutlinedTrailingEl.querySelector('.mdc-text-field__input').required = target.checked;
+
+          tfBoxLeadingEl.querySelector('.mdc-text-field__input').pattern = target.checked ? ".{8,}" : "*";
```

Comment:
`*` is not a valid regex; we should probably just set it to an empty string instead (ditto below).

Also, I think we normally use single quotes instead of double quotes (ditto below).

Reply:
That would mean the input is valid iff it matches the empty string, so I changed it to `.*` instead.